src/config/mod.rs

```diff
@@ -154,33 +166,44 @@ impl Config {
         Ok(())
     }
 
-    pub fn validate(&self) -> Option<Vec<ValidationError>> {
-        let mut messages = vec![];
+    pub fn validate(&self) -> Option<ValidationIssues> {
+        let mut errors = vec![];
+        let mut warnings = vec![];
 
         if self.api_key.is_empty() {
-            messages.push(ValidationError {
+            errors.push(ValidationError {
                 message: "The API key is missing.".to_string(),
                 help_message: formatdoc!(
                     r#"
                         Please either set the `api_key` setting in the configuration file
                         (e.g., run "{} config edit"), pass it as the `--api-key` flag, or
-                        supply the API key via the `CT_API_KEY` environment variable."#,
+                        supply the API key via the `{}` environment variable."#,
+                    CT_API_KEY,
                     binary_name()
                 ),
             });
         }
 
-        if messages.is_empty() {
+        if ConfigEnv::get_override(CT_OLD_API_KEY).is_some()
```

Comment:
This won't handle anyone using `CT_SERVER_URL`, but that setting isn't really published, so I think that's fine.